### ISSUE TITLE
Is there a better way to call two async functions in request preHandler hook?

### ISSUE THREAD

**Issue by varunkamra** [fastify/fastify#4595]
Assume that I have two classes A and B and each class has one async function each.

```js
export class A {
   public static async handle(request, reply) {
     // content
   }
}

export class B {
  public static async handle(request, reply) {
     // content
   }
}
````

I want to call both of these functions in the `preHandler` hook of a request. Currently I am doing something like this:

```js
preHandler: async (request, reply) => {
   await A.handle(request, reply);
   await B.handle(request, reply);
}
````

## 💬  Question 

Is this the best way to call two async functions in the preHandler or is there a better cleaner way of calling these two functions in the preHandler? I checked the preHandler definition and that it only accepts either an array of synchronous functions that can be called or a single async function as it's value, is that right?

## Your Environment

- *node version*: 18
- *fastify version*: >=4.11.0
- *os*: Linux

**Comment by mcollina** [comment]
Thanks for reporting! Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by varunkamra** [comment]
Great thanks! I tried your way too:

```js
preHandler: [A.handle, B.handle]
```

And I kept getting error “calling handle on undefined” for B.

**Comment by mcollina** [comment]
Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by varunkamra** [comment]
I will try to get steps to reproduce this issue.

**Comment by aradwann** [comment]
if it's typing, I think it's related to #4422

**Comment by varunkamra** [comment]
> Great thanks! I tried your way too:
> 
> ```js
> preHandler: [A.handle, B.handle]
> ```
> 
> And I kept getting error “calling handle on undefined” for B.

I am not able to reproduce this issue when I created a separate sample test project, this was specifically happening on jest unit tests in my work project but in the sample project it's not reproducable. @mcollina @aradwann I believe this issue can be closed if it is fixed in #4376.

**Comment by climba03003** [comment]
Close it for now since it is a duplicated issue.
If you believe it is not a duplicate issue, please provide more context and example code.
